### DISCUSSION TITLE
How does ODIS identify repeat queries?

### DISCUSSION THREAD

**Discussion by therealharpaljadeja** [celo-org/social-connect#250] [Help (Q&A)]
![image](https://user-images.githubusercontent.com/38040789/224718124-73891187-4470-4785-9056-441393d46ccf.png)

I don’t understand how ODIS can identify repeat queries 😅
DEK will remain the same until the user changes it, so is there a time period after which ODIS forgets and can’t recognize repeat queries?

**Comment by eelanagaraj** [comment]
Hey so what's actually going on here is that when you use a DEK to generate the blinded message, it functions like a random seed, causing the same plaintext messages to return the same blinded messages. 
ODIS attempts to charge quota once per account, per message (for PNP/LEGACY_PNP), so if the blinded message looks the same, it should count this once without double-charging quota. If you use a random seed to blind a message, the second blinding will result in a "new" message (from the perspective of ODIS).

**Reply by therealharpaljadeja** [reply]
For example:- 

DEK: 0x483945... was used to blind +1234567890

Now, whenever there is an operation of lookup for this number or even registration to add more accounts under the same identifier, ODIS won't charge any quota except for the first time?

**Reply by therealharpaljadeja** [reply]
@eelanagaraj ☝️
